### ISSUE TITLE
Raising uninstantiated exceptions with required arguments

### ISSUE THREAD

**Issue by layday** [microsoft/pyright#986]
Python allows raising exception classes which are instantiated implicitly.  The following is valid Python (for better or worse):

```py
raise Exception
```

This only works for exceptions which can be called without arguments.  Assume that we have a custom exception class:

```py
class FooException(Exception):
    def __init__(self, arg):
        pass
```

... where `arg` is required, then:

```py
raise FooException
```

... will produce a `TypeError` since Python will call `FooException` without arguments.  This is not detected by Pyright.

**Comment by erictraut** [comment]
Thanks for the suggestion. This functionality will be included in the next version of Pyright and Pylance.

**Comment by erictraut** [comment]
This is now addressed in Pyright 1.1.66, which I just published. The change will also be in the next release of Pylance.
